processor/src/main/java/org/mapstruct/ap/internal/conversion/Conversions.java

```diff
@@ -324,10 +330,14 @@ private void register(String sourceTypeName, Class<?> targetClass, ConversionPro
     }
 
     public ConversionProvider getConversion(Type sourceType, Type targetType) {
-        if ( sourceType.isEnumType() && targetType.equals( stringType ) ) {
+        if ( sourceType.isEnumType() &&
+                ( targetType.equals( stringType ) || targetType.equals( integerType ) || targetType.equals( intType ) )
+        ) {
             sourceType = enumType;
         }
-        else if ( targetType.isEnumType() && sourceType.equals( stringType ) ) {
+        else if ( targetType.isEnumType() &&
+                ( sourceType.equals( stringType ) || sourceType.equals( integerType ) || sourceType.equals( intType ) )
```

Comment:
You don't need the `intType` here you can do `sourceType.getBoxedEquivalent().equals( integerType )` instead

Reply:
Thank you very much again for the feedback @filiphr, I implemented the requested change.

Please do not hesitate to contact me to make any changes you deem appropriate.